sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,161 @@
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public LocationPollingStrategy(HttpPipeline httpPipeline, Context context) {
+        this(httpPipeline, context, new DefaultJsonSerializer());
+    }
+
+    /**
+     * Creates an instance of the location polling strategy.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     * @param serializer a custom serializer for serializing and deserializing polling responses
+     */
+    public LocationPollingStrategy(HttpPipeline httpPipeline, Context context, ObjectSerializer serializer) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+        this.serializer = serializer;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        HttpHeader locationHeader = initialResponse.getHeaders().get(PollingConstants.LOCATION);
+        if (locationHeader != null) {
+            try {
+                URL url = new URL(locationHeader.getValue());
+                return Mono.just(true);
+            } catch (MalformedURLException e) {
+                return Mono.just(false);
+            }
+        }
+        return Mono.just(false);
+    }
+
+    @Override
+    public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                                   TypeReference<T> pollResponseType) {
+        HttpHeader locationHeader = response.getHeaders().get(PollingConstants.LOCATION);
+        if (locationHeader != null) {
+            pollingContext.setData(PollingConstants.LOCATION, locationHeader.getValue());
+        }
+        pollingContext.setData(PollingConstants.HTTP_METHOD, response.getRequest().getHttpMethod().name());
+        pollingContext.setData(PollingConstants.REQUEST_URL, response.getRequest().getUrl().toString());
+
+        if (response.getStatusCode() == 200
+                || response.getStatusCode() == 201
+                || response.getStatusCode() == 202
+                || response.getStatusCode() == 204) {
+            String retryAfterValue = response.getHeaders().getValue(PollingConstants.RETRY_AFTER);
+            Duration retryAfter = retryAfterValue == null ? null : Duration.ofSeconds(Long.parseLong(retryAfterValue));
+            return PollingUtils.convertResponse(response.getValue(), serializer, pollResponseType)
+                .map(value -> new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, value, retryAfter))
+                .switchIfEmpty(Mono.defer(() -> Mono.just(new PollResponse<>(
+                    LongRunningOperationStatus.IN_PROGRESS, null, retryAfter))));
+        } else {
+            return Mono.error(new AzureException("Operation failed or cancelled: " + response.getStatusCode()));
```

Comment:
Can we log any other information here if the operation failed? Just logging the status code may not be sufficient to troubleshoot issues. Can we log the response body that contains more details about the failure?

Reply:
Added information of the `location` header (`Operation-Location` header for `OperationResourcePollingStrategy`), and the response body.